backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

```diff
@@ -1330,30 +1328,28 @@ private final void updateBootOrder(List<VmDevice> ... bootableDevices) {
     }
 
     void writeVGpu() {
-        for (String mdevType : MDevTypesUtils.getMDevTypes(vm)) {
+        for (Map<String, Object> mdevSpecParams : MDevTypesUtils.getMdevs(vm, vmInfoBuildUtils.getVmDevices(vm.getId()))) {
             writer.writeStartElement("hostdev");
             writer.writeAttributeString("mode", "subsystem");
             writer.writeAttributeString("type", "mdev");
             writer.writeAttributeString("model", "vfio-pci");
-            if (mdevDisplayOn) {
-                // Nvidia vGPU VNC console is only supported on RHEL >= 7.6
-                // See https://bugzilla.redhat.com/show_bug.cgi?id=1633623 for details and discussion
+            if (!(Boolean)mdevSpecParams.get(MDevTypesUtils.NODISPLAY)) {
                 writer.writeAttributeString("display", "on");
                 if (FeatureSupported.isVgpuFramebufferSupported(vm.getCompatibilityVersion())) {
                     writer.writeAttributeString("ramfb", "on");
                 }
             }
 
+            String address = (String)mdevSpecParams.get(MDevTypesUtils.DEVICE_ID);
             writer.writeStartElement("source");
-            String address = Guid.newGuid().toString();
```

Comment:
unrelated (to this PR) question - so we had the device in the database so it's a managed device, shouldn't we also write its alias?

Reply:
We don't need an alias on the host -- the VM device is fake and we use a host device in the domain XML.  If having an alias is useful for other purposes, it can be added.

Reply:
what I meant to write is: if we have the device in the database then it's a managed device ...
and what I was concerned about is that if we are unable to correlate the device in the database with a reported device (which, as you explained, we won't have) then unless we tweak VmDevicesConverter, the former will be set as unplugged . I looked again at the changes in this PR and I don't see some functionality that would be impacted (because unplugged vgpu devices are not filtered out) so I guess they would just be displayed as unplugged in the VM devices tab.. all that under the assumption that when the aforementioned correlation fails, that device will also be set as unplugged

Reply:
The current version of the patch should keep displaying the device as plugged.

Reply:
cool, I forgot that the correlation for unmanaged host devices is based on address